ament_cmake_auto/cmake/ament_auto_package.cmake

```diff
@@ -61,8 +71,20 @@ macro(ament_auto_package)
 
   # export and install include directory of this package if it has one
   if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/include")
-    ament_export_include_directories("include")
-    install(DIRECTORY include/ DESTINATION include)
+    if(_ARG_AMENT_AUTO_PACKAGE_USE_SCOPED_HEADER_INSTALL_DIR)
+      ament_export_include_directories("include/${PROJECT_NAME}")
+      install(DIRECTORY include/ DESTINATION include/${PROJECT_NAME})
+    else()
+      ament_export_include_directories("include")
+      install(DIRECTORY include/ DESTINATION include)
+      message(WARNING
+        "In this package, headers install destination is set to `include` "
+        "by ament_auto_package. It is recommended to install "
+        "`include/${PROJECT_NAME}` instead and will be the default behavior "
+        "of ament_auto_package from ROS 2 Kilted Kaiju. On distributions before "
+        "Kilted, ament_auto_package behaves the same way when you use "
+        "USE_SCOPED_HEADER_INSTALL_DIR option.")
```

Comment:
Please use `NOTICE` instead of `WARNING`.

CMake warnings are often reported by CI systems, like our [Humble CI jobs on build.ros2.org](https://build.ros2.org/view/Hci/). However, it would be reasonable for a maintainer to decide to not specify `USE_SCOPED_HEADER_INSTALL_DIR` in Humble and Jazzy to avoid the include directory path change affecting their non-CMake users. These maintainers would then be stuck with their jobs reporting warnings they have already decided not to pay attention to.

Reply:
Thank you for your feedback!
I simply deleted `WARNING` mode in [`9e6f3ea` (#577)](https://github.com/ament/ament_cmake/pull/577/commits/9e6f3ea2f876f22e9bf870f28adf4f21d3a84ca2), because `NOTICE` mode is introduced from CMake 3.15, that is greater than 3.12 (cmake minimum required version of this project).
According to [CMake documentation](https://cmake.org/cmake/help/v3.15/command/message.html#:~:text=else%20no%20message.-,(none)%20or%20NOTICE,-Important%20message%20printed), no mode in message function has same behavior to `NOTICE` mode.